packages/db-model/src/model.ts

```diff
@@ -161,67 +169,72 @@ export function createBaseModel(resolver?: ConnectionResolver) {
      * @param {Transaction} [trx]
      * @returns {Promise<T[]>}
      */
-    public static insert<T>(data: object | object[], trx?: Transaction): Promise<T[]> {
+    public static insert<T>(data: object | object[], trx?: Knex.Transaction): Promise<T[]> {
       return db.insert<T>(this.getConnection(), this.table, data, trx);
     }
 
     /**
-     * Update records by id.
+     * Update records by primary key.
      *
-     * @param {number} id
-     * @param {object} params
+     * @param {string} pk
+     * @param {any} params
      * @param {Transaction} transaction
-     * @returns {Promise<object>}
+     * @returns {Knex.QueryBuilder}
      */
-    public static updateById<T>(id: number | number[], params: object, trx?: Transaction): Promise<T[]> {
-      return db.updateById<T>(this.getConnection(), this.table, id, params, trx);
+    public static updateByPk<T>(pk: string, params: object, trx?: Knex.Transaction): Promise<T[]> {
+      const pkParams: { [key: string]: any } = {};
+      pkParams[this.pk] = pk;
```

Comment:
I think this should work. 
```suggestion
      const pkParams = { [this.pk]: pk };
 ```